src/diag/diag_process.cpp

```diff
@@ -45,8 +45,9 @@ namespace ot {
 namespace Diagnostics {
 
 const struct Diag::Command Diag::sCommands[] = {
```

Comment:
I think we don't need the last `{NULL, NULL}` if we use `OT_ARRAY_LENTH()` to compute the number of items. Would you like to submit another PR to enhance this?

Reply:
Sure, will update in another PR.